### PR TITLE
Pubmatic Bid Adapter: add support for JW Player

### PR DESCRIPTION
**Pull request by pm-manasi-moghe**
## Type of change

- [ ] Feature

## Description of change
Adding support for JWPlayer in pubmaticBidAdapter.js

### CODE REVIEW

**Comment by patmmccann** [comment]
@karimMourra do you want to take a look?

**Review by ChrisHuie — APPROVED** [review]

**Comment by pm-manasi-moghe** [comment]
Any updates on when this PR will be merged?

**Comment by patmmccann** [comment]
It won't be merged before review. Also @karimMourra this highlights the issue we were discussing. JW should write its segments to the Ortb2 object so that adapters don't need to develop tooling to it.

**Comment by pm-manasi-moghe** [comment]
Thanks @patmmccann, when do we expect the review to happen?

**Comment by karimMourra** [comment]
@patmmccann I see your point about writing to the Ortb2 object. WE should discuss with the other RTD contributors to ensure consistency.

**Comment by mike-chowla** [comment]
@karimMourra @patmmccann Not clear on what the plan is here with the Ortb2 object.  Is that future item for JW to take care off or something that needs to be done ahead of this PR?  We (PubMatic) would like to get our integration working with the JW data module as soon as possible.

**Comment by patmmccann** [comment]
not trying to delay this PR, just highlighting it as something that shouldnt need to be done if the rtd module was writing data to the new place

@karimMourra we do have a group to coordinate this with meetings every week. What coordination do you want to achieve?

File: modules/pubmaticBidAdapter.js [hunk 1]
```diff
@@ -605,6 +605,23 @@ function _addDealCustomTargetings(imp, bid) {
...
+
+  jwPlayerData += 'jw-id=' + jwSegData.content.id + '|'; // add the content id first
+
+  for (var i = 0; i < maxLength; i++) {
+    jwPlayerData += 'jw-' + jwSegData.segments[i] + '=1|';
+  }
+  (imp.ext && imp.ext.key_val == undefined)
+    ? imp.ext.key_val = jwPlayerData.substring(0, jwPlayerData.length - 1)
+    : imp.ext.key_val += '|' + jwPlayerData.substring(0, jwPlayerData.length - 1);
+}
```
[the remark this answers is not in the document]
Comment (pm-manasi-moghe) [reply modules/pubmaticBidAdapter.js]: Done.

File: modules/pubmaticBidAdapter.js [hunk 2]
```diff
@@ -605,6 +605,23 @@ function _addDealCustomTargetings(imp, bid) {
...
+  if (jwSegData === undefined || jwSegData === '' || !jwSegData.hasOwnProperty('segments')) return;
+
+  var maxLength = jwSegData.segments.length;
+
+  jwPlayerData += 'jw-id=' + jwSegData.content.id + '|'; // add the content id first
+
+  for (var i = 0; i < maxLength; i++) {
+    jwPlayerData += 'jw-' + jwSegData.segments[i] + '=1|';
+  }
+  (imp.ext && imp.ext.key_val == undefined)
```
[the remark this answers is not in the document]
Comment (pm-manasi-moghe) [reply modules/pubmaticBidAdapter.js]: Fixed.

**Review by ncolletti — APPROVED** [review]
lgtm

**Comment by pm-manasi-moghe** [comment]
Any updates on the review progress?

**Review by karimMourra — APPROVED** [review]

**Comment by karimMourra** [comment]
LGTM!

**Comment by karimMourra** [comment]
@patmmccann the RTD module was led by the Tools committee I believe, with Browsi at the helm but i think a github issue would suffice. We would want all RTD submodules to write to Ortb2; this could potentially be done in one place by the core module.

**Comment by patmmccann** [comment]
@karimMourra the audigent and sirdata and many other rtd modules are already doing this; audigent is suggesting their reference implementation as the solve to https://github.com/prebid/Prebid.js/issues/6425

I don't think any work is required on the main modules for jw to follow suit and write to the main object similarly from their submodule.

**Pull request merged**